solr/core/src/java/org/apache/solr/api/AnnotatedApi.java

```diff
@@ -113,7 +116,7 @@ public EndPoint getEndPoint() {
       return Collections.singletonList(new AnnotatedApi(specProvider, endPoint, commands, null));
     } else {
       List<Api> apis = new ArrayList<>();
-      for (Method m : klas.getDeclaredMethods()) {
+      for (Method m : klas.getMethods()) {
         EndPoint endPoint = m.getAnnotation(EndPoint.class);
         if (endPoint == null) continue;
         if (!Modifier.isPublic(m.getModifiers())) {
```

Comment:
Is this still needed? Wouldn't `MethodHandles.publicLookup().accessClass(theClass)` only return `public` methods?

Reply:
Yes, we still need some way to check public methods. The accessClass() method above returns itsself, but the check for public is not needed if we use getMethods() instead of getDeclaredMethods().
The accessClass() check can't be backported to 8.x as it is new in Java 9. So we need to somehow do the check for accessibility manually.

Reply:
So you agree @uschindler that the `if` can be removed in the `master` branch we're looking at, did I understand you correctly?

Reply:
Yes. For both branches.

In 8.x we have to fix the class accessibility check (unrelated to the if statement).

Reply:
The if statement is not needed, because the public checks are done by unreflect().

This code is from reflection times to catch error early. By using method handles everything is checked and linked early, so you can't get any surprises later. So the public checks is complete obsolete, all this is handled by unreflect.